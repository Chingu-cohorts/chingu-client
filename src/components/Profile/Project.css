.project {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.project .project-container {
  margin-top: 1rem;
}

.project .project-container .row {
  margin-bottom: 0;
}

.project img {
  max-width: 100%;
  max-height: 100%;
  padding: 0.5rem;
}

.project ul li {
  display: inline-block;
  margin-right: 1rem;
}

.project strong {
  font-weight: 600;
}
