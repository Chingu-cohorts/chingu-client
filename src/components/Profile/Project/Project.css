.project {
  margin-bottom: 1rem;
}

.project img {
  max-width: 100%;
  max-height: 100%;
}

.project ul li {
  display: inline-block;
  margin-right: 1rem;
}

.project strong {
  font-weight: 600;
}
