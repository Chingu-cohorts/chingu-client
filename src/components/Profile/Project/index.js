import React from "react";
import { Card } from "antd";

import "./Project.css";

const Project = props => (
  <Card>
    <div className="image">
      <img
        alt="example"
        width="100%"
        src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png"
      />
    </div>
    <div className="content">
      <h3>{props.title}</h3>
      <p>{props.description}</p>
      <p>
        <ul>
          <li>
            {props.github_url ? (
              <a href={props.github_url}>Github</a>
            ) : (
              "No GitHub"
            )}
          </li>
          <li>
            {props.project_url ? (
              <a href={props.project_url}>Live</a>
            ) : (
              "No Project URL"
            )}
          </li>
        </ul>
      </p>
      <p>
        <strong>Team</strong>
        <ul>
          {props.users.map(user => {
            return <li>{user.username}</li>;
          })}
        </ul>
      </p>
    </div>
  </Card>
);

export default Project;
