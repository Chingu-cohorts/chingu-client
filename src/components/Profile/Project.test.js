import React from "react";
import { shallow } from "enzyme";

import Project from "./Project";

describe("Project Component", () => {
  const props = {
    image_url: "https://lorempixel.com/nature/200/200",
    title: "Chingu Developers Network",
    description: "An amazing project",
    github_url: "https://github.com/chingu-x/chingu-client",
    users: [
      {
        username: "Oxyrus"
      },
      {
        username: "Vampiire"
      },
      {
        username: "TheSabby"
      }
    ]
  };
  it("renders without crashing", () => {
    const wrapper = shallow(<Project {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
