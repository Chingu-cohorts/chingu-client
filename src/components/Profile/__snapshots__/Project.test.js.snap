// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Project
    description="An amazing project"
    github_url="https://github.com/chingu-x/chingu-client"
    image_url="https://lorempixel.com/nature/200/200"
    title="Chingu Developers Network"
    users={
        Array [
            Object {
              "username": "Oxyrus",
            },
            Object {
              "username": "Vampiire",
            },
            Object {
              "username": "TheSabby",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="image"
>
          <img
                    alt="example"
                    src="https://lorempixel.com/nature/200/200"
                    width="100%"
          />
</div>,
        <div
          className="content"
>
          <h3>
                    Chingu Developers Network
          </h3>
          <p>
                    An amazing project
          </p>
          <ul>
                    <li>
                              <a
                                        href="https://github.com/chingu-x/chingu-client"
                              >
                                        Github
                              </a>
                    </li>
                    <li>
                              No Project URL
                    </li>
          </ul>
          <strong>
                    Team
          </strong>
          <ul>
                    <li>
                              Oxyrus
                    </li>
                    <li>
                              Vampiire
                    </li>
                    <li>
                              TheSabby
                    </li>
          </ul>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="example"
            src="https://lorempixel.com/nature/200/200"
            width="100%"
/>,
          "className": "image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "example",
            "src": "https://lorempixel.com/nature/200/200",
            "width": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Chingu Developers Network
</h3>,
            <p>
              An amazing project
</p>,
            <ul>
              <li>
                            <a
                                          href="https://github.com/chingu-x/chingu-client"
                            >
                                          Github
                            </a>
              </li>
              <li>
                            No Project URL
              </li>
</ul>,
            <strong>
              Team
</strong>,
            <ul>
              <li>
                            Oxyrus
              </li>
              <li>
                            Vampiire
              </li>
              <li>
                            TheSabby
              </li>
</ul>,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Chingu Developers Network",
            },
            "ref": null,
            "rendered": "Chingu Developers Network",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "An amazing project",
            },
            "ref": null,
            "rendered": "An amazing project",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <a
                                    href="https://github.com/chingu-x/chingu-client"
                  >
                                    Github
                  </a>
</li>,
                <li>
                  No Project URL
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="https://github.com/chingu-x/chingu-client"
>
                    Github
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Github",
                    "href": "https://github.com/chingu-x/chingu-client",
                  },
                  "ref": null,
                  "rendered": "Github",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No Project URL",
                },
                "ref": null,
                "rendered": "No Project URL",
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Team",
            },
            "ref": null,
            "rendered": "Team",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Oxyrus
</li>,
                <li>
                  Vampiire
</li>,
                <li>
                  TheSabby
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Oxyrus",
                "nodeType": "host",
                "props": Object {
                  "children": "Oxyrus",
                },
                "ref": null,
                "rendered": "Oxyrus",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "Vampiire",
                "nodeType": "host",
                "props": Object {
                  "children": "Vampiire",
                },
                "ref": null,
                "rendered": "Vampiire",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "TheSabby",
                "nodeType": "host",
                "props": Object {
                  "children": "TheSabby",
                },
                "ref": null,
                "rendered": "TheSabby",
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="image"
>
            <img
                        alt="example"
                        src="https://lorempixel.com/nature/200/200"
                        width="100%"
            />
</div>,
          <div
            className="content"
>
            <h3>
                        Chingu Developers Network
            </h3>
            <p>
                        An amazing project
            </p>
            <ul>
                        <li>
                                    <a
                                                href="https://github.com/chingu-x/chingu-client"
                                    >
                                                Github
                                    </a>
                        </li>
                        <li>
                                    No Project URL
                        </li>
            </ul>
            <strong>
                        Team
            </strong>
            <ul>
                        <li>
                                    Oxyrus
                        </li>
                        <li>
                                    Vampiire
                        </li>
                        <li>
                                    TheSabby
                        </li>
            </ul>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="example"
              src="https://lorempixel.com/nature/200/200"
              width="100%"
/>,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "example",
              "src": "https://lorempixel.com/nature/200/200",
              "width": "100%",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Chingu Developers Network
</h3>,
              <p>
                An amazing project
</p>,
              <ul>
                <li>
                                <a
                                                href="https://github.com/chingu-x/chingu-client"
                                >
                                                Github
                                </a>
                </li>
                <li>
                                No Project URL
                </li>
</ul>,
              <strong>
                Team
</strong>,
              <ul>
                <li>
                                Oxyrus
                </li>
                <li>
                                Vampiire
                </li>
                <li>
                                TheSabby
                </li>
</ul>,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Chingu Developers Network",
              },
              "ref": null,
              "rendered": "Chingu Developers Network",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "An amazing project",
              },
              "ref": null,
              "rendered": "An amazing project",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <a
                                        href="https://github.com/chingu-x/chingu-client"
                    >
                                        Github
                    </a>
</li>,
                  <li>
                    No Project URL
</li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="https://github.com/chingu-x/chingu-client"
>
                      Github
</a>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Github",
                      "href": "https://github.com/chingu-x/chingu-client",
                    },
                    "ref": null,
                    "rendered": "Github",
                    "type": "a",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No Project URL",
                  },
                  "ref": null,
                  "rendered": "No Project URL",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Team",
              },
              "ref": null,
              "rendered": "Team",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    Oxyrus
</li>,
                  <li>
                    Vampiire
</li>,
                  <li>
                    TheSabby
</li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Oxyrus",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Oxyrus",
                  },
                  "ref": null,
                  "rendered": "Oxyrus",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "Vampiire",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vampiire",
                  },
                  "ref": null,
                  "rendered": "Vampiire",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "TheSabby",
                  "nodeType": "host",
                  "props": Object {
                    "children": "TheSabby",
                  },
                  "ref": null,
                  "rendered": "TheSabby",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
